cmake/AwsCFlags.cmake

```diff
@@ -8,6 +8,7 @@ include(CMakeParseArguments) # needed for CMake v3.4 and lower
 option(AWS_ENABLE_LTO "Enables LTO on libraries. Ensure this is set on all consumed targets, or linking will fail" OFF)
 option(LEGACY_COMPILER_SUPPORT "This enables builds with compiler versions such as gcc 4.1.2. This is not a 'supported' feature; it's just a best effort." OFF)
 option(AWS_SUPPORT_WIN7 "Restricts WINAPI calls to Win7 and older (This will have implications in downstream libraries that use TLS especially)" OFF)
+option(WARNINGS_ARE_ERRORS "Compiler warning is treated as an error. Try turning this off when observing errors on a new or unpopular compiler" ON)
```

Comment:
trivial: should we namespace this option with "AWS_"?
AWS_WARNINGS_ARE_ERRORS?

Reply:
Makes sense. It is a very general of a name that could clash with another project's naming